File: LC/diffTwoArrays.ts
const diffTwoArrays = (nums1: number[], nums2: number[]): number[][] => {
  const set1 = new Set(nums1);
  const set2 = new Set(nums2);

  const ans1 = [...set1].filter(n => !set2.has(n));
  const ans2 = [...set2].filter(n => !set1.has(n));

  return [ ans1, ans2 ];
}

const n1 = [1,2,3];
const n2 = [2,4,6];

const m1 = [1,2,3,3];
const m2 = [1,1,2,2];

console.log(diffTwoArrays(n1, n2));
console.log(diffTwoArrays(m1, m2));

File: LC/diffTwoArrays.rb
def diff_two_arrays(nums1, nums2)
  a, b = nums1.uniq, nums2.uniq
  [a-b, b-a]
end

n_1 = [1,2,3]
m_1 = [2,4,6]

p diff_two_arrays(n_1, m_1)